source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +168,141 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+std::string TextBlock::ParseISO8601UsingRegex() const
+{
+	std::regex pattern("\\{\\{(DATE|Date|TIME|Time)\\((\\d{4})-{1}(\\d{2})-{1}(\\d{2})T(\\d{2}):{1}(\\d{2}):{1}(\\d{2})(Z|([+-])(\\d{2}):{1}(\\d{2}))(\\)|,Short\\)|,SHORT\\)|,Long\\)|,LONG\\))\\}\\}");
```

Comment:
>(DATE|Date|TIME|Time [](start = 27, length = 20)

This will fail for {{daTE(2017-02-14T06:08:39Z,Short)}} which is valid in the typescript renderer/visualizer

Reply:
i think math said he is aiming for stricker standard.
let me check with mathh.

---
In reply to: [149181253](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149181253) [](ancestors = 149181253)